### ISSUE TITLE
Crash after update to Xcode 13.2.1

### ISSUE THREAD

**Issue by TomQv** [MichaelRumpler/GestureSample#38]

**Comment by TomQv** [comment]
I updated my dev-mashine at the same time to Xcode 13.2.1, macOS 12.2.1 and Visual Studio Mac 8.10.19.
Using MR.Gesture 2.2.0.
Now when starting ios-app with MR.Gersture I receive the following error:

`[UIPanGestureRecognizer setAllowedScrollTypesMask:]: unrecognized selector sent to instance 0x117b18f10`

I'm using these events and I have to remove all these line to get my app started:

```
gesturecontainer.Pinching += Gesturecontainer_Pinching;
gesturecontainer.Pinched += Gesturecontainer_Pinched;
gesturecontainer.Panning += Gesturecontainer_Panning;
gesturecontainer.Panned += Gesturecontainer_Panned;
gesturecontainer.Tapped += Gesturecontainer_Tapped;
gesturecontainer.LongPressed += Gesturecontainer_LongPressed;
gesturecontainer.ScrollWheelChanged += Gesturecontainer_ScrollWheelChanged;

```

Any ideas? Unfortunate I don't know anymore, which versions of Xcode, VS and macOS I had before, but all was fine before.

**Comment by TomQv** [comment]
If I only remove ScrollWheelChanged, which anyway doesn't make sense on ios, the app starts. But when I just tap, which triggers Tapped-event, then this error comes:

`[MR_Gestures_iOS_DownUpGestureRecognizer buttonMask]: unrecognized selector sent to instance 0x283c18ff0`

**Comment by TomQv** [comment]
I downgraded to MR.Gesture 2.1.3 and all is fine.
Of course without ScrollWheelChanged

**Comment by MichaelRumpler** [comment]
That sounds more like an iOS version problem. What iOS version does your device have?

You can use ScrollWheelChanged just as well on iOS if you connect a bluetooth mouse to your iPad.

**Comment by TomQv** [comment]
Its 14.8 on the iPhone and it was also not updated recently.

As said, issue started with Xcode and macOS update on dev-mashine. 
So now only MR 2.1.3 works, MR 2.2.0 gives the crash

**Comment by TomQv** [comment]
I think I found the issue. I had other problems too with the app and finally I found this issue:
https://github.com/xamarin/Xamarin.Forms/issues/15104
Following their workaround going back to Xamarin.Forms 5.0.0.2291 solved it. 
MR.Gesture 2.2.0 is now also fine.

**Comment by MichaelRumpler** [comment]
That bug is a `NullReferenceException` in the `ImageRenderer`. You neither mentioned a NRE here nor did the other issue have a

    [UIPanGestureRecognizer setAllowedScrollTypesMask:]: unrecognized selector sent to instance

or something similar.

**Comment by TomQv** [comment]
I said, that MR.Gesture 2.2.0 is having issues with XF 5.0.0.2337, and with XF 5.0.0.2291 not.
I didn't try to figure out why exactly, but at least I wanted to give you the information in case somebody else has similar issues.

**Comment by MichaelRumpler** [comment]
I already had Xcode 13.2.1 on my Mac, so I only tested with that version.

| Device | XF | |
| --- | --- | --- |
| iPad 15.2.1 | 5.0.0.2012 | works |
| iPhone Sim 15.2 | 5.0.0.2012 | works |
| iPad 15.2.1 | 5.0.0.2337 | works |
| iPhone Sim 15.2 | 5.0.0.2337 | works |

I don't have a physical iPhone anymore, so I tested in the iPhone Simulator. Maybe that one behaves differently?

However, I cannot reproduce it. Can you send me a repro project?

The error message you pasted

    [UIPanGestureRecognizer setAllowedScrollTypesMask:]: unrecognized selector sent to instance 0x117b18f10

comes from something I added in MR.Gestures 2.2.0 to handle the `ScrollWheelChanged` event. According to [Apples docs](https://developer.apple.com/documentation/uikit/uipangesturerecognizer/3538978-allowedscrolltypesmask) it should work from iOS 13.4, so your 14.8 should be covered. But it makes no sense that the error is gone when you go back to an older XF version.

**Comment by TomQv** [comment]
> > But it makes no sense that the error is gone when you go back to an older XF version.

Yes, but I gave up on asking this long time ago. The whole vs-mac/mono/xamarin environment is so fragile, and its not the first time we had to go back to an older version after spending hours and days searching in own code for the bug. 
Unfortunate I can't provide a test. This is part of a big project and I'm already short of time.
But thanks for your support, much appreciated.
